app/controlplane/integrations/dependencytrack/cyclonedx/v1/integration.go

```diff
@@ -16,34 +16,51 @@
 package integration
 
 import (
+	"bytes"
 	"context"
 	"errors"
 	"fmt"
 
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
 	core "github.com/chainloop-dev/chainloop/app/controlplane/integrations"
 	"github.com/chainloop-dev/chainloop/app/controlplane/integrations/dependencytrack/cyclonedx/v1/uploader"
 	pb "github.com/chainloop-dev/chainloop/app/controlplane/integrations/gen/dependencytrack/cyclonedx/v1"
+	"github.com/go-kratos/kratos/v2/log"
 	"google.golang.org/protobuf/types/known/anypb"
 )
 
-const Kind = "Dependency-Track"
+const ID = "dependencytrack.cyclonedx.v1"
+const description = "Dependency Track CycloneDX Software Bill Of Materials Integration"
 
-var _ core.Registrable = (*Integration)(nil)
-var _ core.Attachable = (*Integration)(nil)
+var _ core.FanOut = (*DependencyTrack)(nil)
 
-type Integration struct {
-	ID string
+type DependencyTrack struct {
+	*core.BaseIntegration
 }
 
 // Attach attaches the integration service to the given grpc server.
 // In the future this will be a plugin entrypoint
-func NewIntegration() (*Integration, error) {
-	return &Integration{
-		ID: Kind,
+func NewIntegration(l log.Logger) (*DependencyTrack, error) {
+	base, err := core.NewBaseIntegration(
+		ID, description,
+		core.WithInputMaterial(schemaapi.CraftingSchema_Material_SBOM_CYCLONEDX_JSON),
```

Comment:
the dependency-track integration now chooses what kind of material they want. 

This is not being used yet, but it will be the way the future dispatcher will know what each integration cares about